packages/front-end/components/DataViz/DataVizFilterPanel.tsx

```diff
@@ -0,0 +1,231 @@
+      dataVizConfig,
+      sampleRow,
+    );
+    setColumnFilterOptions(columnFilterOptions);
+  }, [dataVizConfig, sampleRow]);
+
+  const filters = dataVizConfig.filter || [];
+
+  // Early return if no column filter options are available
+  if (!columnFilterOptions.length) return null;
+
+  if (!dataVizConfig.chartType) return null;
+
+  return (
+    <>
+      <Flex
+        direction="column"
+        height="100%"
+        style={{
+          border: "1px solid var(--gray-a3)",
+          borderRadius: "var(--radius-4)",
+          overflow: "hidden",
+          backgroundColor: "var(--color-panel-translucent)",
+        }}
+      >
+        <Collapsible
+          open={true}
+          trigger={
+            <div
+              style={{
+                paddingLeft: "12px",
+                paddingRight: "12px",
+                paddingTop: "12px",
+                paddingBottom: "12px",
+                borderBottom: "1px solid var(--gray-a3)",
+              }}
+            >
+              <Text style={{ color: "var(--color-text-mid)", fontWeight: 500 }}>
+                <Flex justify="between" align="center">
+                  <Flex align="center" gap="1">
+                    <PiSlidersHorizontal
+                      style={{
+                        color: "var(--violet-11)",
+                      }}
+                      size={20}
+                    />
+                    Filters
+                    <Badge
+                      label={filters.length.toString()}
+                      color="violet"
+                      radius="full"
+                      variant="soft"
+                    />
+                  </Flex>
+                  <Flex align="center" gap="1">
+                    <Button
+                      variant="ghost"
+                      color="red"
+                      disabled={filters.length === 0}
+                      onClick={() => {
+                        onDataVizConfigChange({
+                          ...dataVizConfig,
+                          filter: [],
+                        });
+                      }}
+                    >
+                      Clear
+                    </Button>
+                    <FaAngleRight className="chevron" />
+                  </Flex>
+                </Flex>
+              </Text>
+            </div>
+          }
+          transitionTime={100}
+        >
+          <Box p="4" height="fit-content">
+            <Flex direction="column" gap="4">
+              {filters.length > 0 &&
+                filters.map((filter, index) => {
+                  return (
+                    <DataVizFilter
+                      key={index}
+                      filterIndex={index}
+                      columnFilterOptions={columnFilterOptions}
+                      dataVizConfig={dataVizConfig}
+                      onDataVizConfigChange={onDataVizConfigChange}
+                      rows={rows}
```

Comment:
question: Should the DataVizFilter component receive the whole dataVizConfig? Or should we pass in just the specific filter we are rendering and keep the 'add to dataVizConfig filters' part in this component?